source/simulator/nullspace.cc

```diff
@@ -103,11 +245,126 @@ namespace aspect
+        x_velocity_mask.set(0, true);
+
+        const bool success = internal::constrain_point_on_boundary_to_zero(dof_handler,
+                                                                           mapping,
+                                                                           location,
+                                                                           x_velocity_mask,
+                                                                           level,
+                                                                           constraints);
+
+        const bool global_success = (Utilities::MPI::max(success?1:0, mpi_communicator) == 1);
+        AssertThrow(level != -1 || global_success,
+                    ExcInternalError("ERROR: constrain_point_on_boundary() could not find the specified support point."));
+      }
+    else if (dim==3)
+      {
+        // point 1 is in positive z direction, constrain x and y:
+        const Point<dim> &location1 = geometry_model.representative_point(point_depth);
+        ComponentMask x_and_y_velocity_mask(dof_handler.get_fe().n_components(), false);
+        x_and_y_velocity_mask.set(0, true);
+        x_and_y_velocity_mask.set(1, true);
+        bool success = internal::constrain_point_on_boundary_to_zero(dof_handler,
+                                                                     mapping,
+                                                                     location1,
+                                                                     x_and_y_velocity_mask,
+                                                                     level,
+                                                                     constraints);
+        bool global_success = (1==Utilities::MPI::max(success?1:0, mpi_communicator));
+        AssertThrow(level != -1 || global_success,
+                    ExcInternalError("ERROR: constrain_point_on_boundary() could not find the specified support point."));
+
+        // point 2 is in positive x direction, constrain y
+        Point<dim> location2;
+        location2[0] = location1[dim-1];
+        ComponentMask y_velocity_mask(dof_handler.get_fe().n_components(), false);
+        y_velocity_mask.set(1, true);
+        success = internal::constrain_point_on_boundary_to_zero(dof_handler,
+                                                                mapping,
+                                                                location2,
+                                                                y_velocity_mask,
+                                                                level,
+                                                                constraints);
+        global_success = (Utilities::MPI::max(success?1:0, mpi_communicator)==1);
+        AssertThrow(level != -1 || global_success,
+                    ExcInternalError("ERROR: constrain_point_on_boundary() could not find the specified support point."));
```

Comment:
```suggestion
                    ExcInternalError("constrain_point_on_boundary() could not find the specified support point."));
```

Reply:
On second thought, the function name here is also no longer correct. Just drop it, the error message can be followed to the place where the call happens.